From 1bb9fa3bb139a28c3596c3d2e44f38a2270aaf08 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:42:42 -0400
Subject: [alsa] hda: Set up initial pins for Acer Aspire V5

Message-id: <140972823573-443-patchset-jkysela@redhat.com>
Patchwork-id: 91132
O-Subject: [RHEL 7.1 PATCH 442/442] ALSA: hda - Set up initial pins for Acer Aspire V5
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Acer Aspire V5 doesn't set up the pins correctly at the cold boot
while the pins are corrected after the warm reboot.  This patch gives
the proper pin configs statically in the driver as a workaround.

Bugzilla: https://bugzilla.kernel.org/show_bug.cgi?id=81561
Signed-off-by: Takashi Iwai <tiwai@suse.de>
Signed-off-by: Jaroslav Kysela <perex@perex.cz>

(picked from kernel: 1a22e7758eabc431d6d8af085dc6e4c5031779a6)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_realtek.c b/sound/pci/hda/patch_realtek.c
index e0fff47..48d6d10 100644
--- a/sound/pci/hda/patch_realtek.c
+++ b/sound/pci/hda/patch_realtek.c
@@ -4353,6 +4353,7 @@ enum {
  ALC292_FIXUP_TPT440_DOCK,
  ALC283_FIXUP_BXBT2807_MIC,
  ALC255_FIXUP_DELL_WMI_MIC_MUTE_LED,
+ ALC282_FIXUP_ASPIRE_V5_PINS,
 };
 
 static const struct hda_fixup alc269_fixups[] = {
@@ -4800,6 +4801,22 @@ static const struct hda_fixup alc269_fixups[] = {
   .chained_before = true,
   .chain_id = ALC255_FIXUP_DELL1_MIC_NO_PRESENCE
  },
+ [ALC282_FIXUP_ASPIRE_V5_PINS] = {
+  .type = HDA_FIXUP_PINS,
+  .v.pins = (const struct hda_pintbl[]) {
+   { 0x12, 0x90a60130 },
+   { 0x14, 0x90170110 },
+   { 0x17, 0x40000008 },
+   { 0x18, 0x411111f0 },
+   { 0x19, 0x411111f0 },
+   { 0x1a, 0x411111f0 },
+   { 0x1b, 0x411111f0 },
+   { 0x1d, 0x40f89b2d },
+   { 0x1e, 0x411111f0 },
+   { 0x21, 0x0321101f },
+   { },
+  },
+ },
 
 };
 
@@ -4811,6 +4828,7 @@ static const struct snd_pci_quirk alc269_fixup_tbl[] = {
  SND_PCI_QUIRK(0x1025, 0x0740, "Acer AO725", ALC271_FIXUP_HP_GATE_MIC_JACK),
  SND_PCI_QUIRK(0x1025, 0x0742, "Acer AO756", ALC271_FIXUP_HP_GATE_MIC_JACK),
  SND_PCI_QUIRK(0x1025, 0x0775, "Acer Aspire E1-572", ALC271_FIXUP_HP_GATE_MIC_JACK_E1_572),
+ SND_PCI_QUIRK(0x1025, 0x079b, "Acer Aspire V5-573G", ALC282_FIXUP_ASPIRE_V5_PINS),
  SND_PCI_QUIRK(0x1028, 0x0470, "Dell M101z", ALC269_FIXUP_DELL_M101Z),
  SND_PCI_QUIRK(0x1028, 0x05bd, "Dell", ALC269_FIXUP_DELL2_MIC_NO_PRESENCE),
  SND_PCI_QUIRK(0x1028, 0x05be, "Dell", ALC269_FIXUP_DELL2_MIC_NO_PRESENCE),
-- 
1.7.1